plugins/woocommerce/includes/wc-template-functions.php

```diff
@@ -1635,6 +1635,10 @@ function woocommerce_catalog_ordering( $attributes = null ) {
 			unset( $catalog_orderby_options['rating'] );
 		}
 
+		if ( is_array( $orderby ) ) {
+			$orderby = current( array_keys( $catalog_orderby_options ) );
```

Comment:
🤔 It's too bad we can't see in the error message what was in the array that was passed here that caused the fatal error. My guess is that it was probably multiple catalog order options, e.g. `array( 'rating', 'menu_order' )`, because it's valid to specify multiple orderby's this way in `WP_Query`.

What if, instead of falling back on the first item in the enum list for catalog orderby options, we picked the first item in the array that was passed? If the first item isn't actually a valid orderby, it would still get fixed in the next conditional check. Better yet, we could do an array intersect first. Something like:

```php
if ( is_array( $orderby ) ) {
	$orderby = current( array_intersect( $orderby, $catalog_orderby_options ) );
}
```

Wdyt?